.github/workflows/lint-and-test.yaml

```diff
@@ -21,15 +23,78 @@ jobs:
             image: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v2
-      - run: pwd && docker-compose --version && docker --version
+      - uses: actions/checkout@v3
+
+      - name: Check version info
+        run: pwd && docker-compose --version && docker --version
+
+      - name: Set up Docker Buildx
+        if: ${{ env.USE_CACHE == 'true' }}
+        id: buildx
+        uses: docker/setup-buildx-action@v2
+        with:
+          driver-opts: network=host
+
+      - name: Cache Docker layers
+        if: ${{ env.USE_CACHE == 'true' }}
+        uses: actions/cache@v3
+        with:
+          path: /tmp/.buildx-cache/${{ matrix.python-version }}
+          key: buildx-${{ github.ref }}-${{ github.sha }}
+          restore-keys: |
+            buildx-${{ github.ref }}
+            buildx-
+
+      - name: Cache Docker registry
+        if: ${{ env.USE_CACHE == 'true' }}
+        uses: actions/cache@v3
+        with:
+          path: /tmp/docker-registry/${{ matrix.python-version }}
+          key: docker-registry-${{ github.ref }}-${{ github.sha }}
+          restore-keys: |
+            docker-registry-${{ github.ref }}
+            docker-registry-
+
+      - name: Boot-up local Docker registry
+        if: ${{ env.USE_CACHE == 'true' }}
+        run: docker run -d -p 5000:5000 --restart=always --name registry -v /tmp/docker-registry:/var/lib/registry registry:2
+      
+      - name: Wait for Docker registry
+        if: ${{ env.USE_CACHE == 'true' }}
+        run: npx wait-on tcp:5000
```

Comment:
I guess it's better to set `--httpTimeout`. The document says `default 0 which results in using the OS default`. It's ok If it's clear what's the default value of `ubuntu-latest` and it's good value. Otherwise we may consume GitHub action time too much.